### PR TITLE
feat: Apple Notes-style inline math with live results

### PR DESCRIPTION
**Pull request by i-doll**
## What

Inline math in the editor, modeled on Apple Notes' math feature: `Name = expression` lines define variables (highlighted in the editor), and a trailing `=` shows the computed result as a read-only ghost decoration after the `=` — recomputed live from the note text and **never written to the note or disk**.

```
Params = 32 Billion
BytesPerParam = 4b # AWQ 4-bit
ModelSize = Params × BytesPerParam + 20% = ⟨19.2 GB⟩
32GB / 3200 NOK = ⟨10 MB/NOK⟩
```

## Language

- Arithmetic: `+ - × ÷ * / ^`, parentheses, unary minus, percentages (`X + 20%` = X×1.2 Apple-style)
- Numbers: decimals, thousands-separated input (`16 000 000 000`), hex/bin literals (`0xFF`, `0b1010`), `k/M/G/T` scale suffixes, word multipliers (`32 Billion`)
- **Units with dimensional analysis**: data (SI + IEC + bits), time, energy, power, length, mass, frequency — `6kJ/s = ⟨6 kW⟩`, `8MB / 2s = ⟨4 MB/s⟩`, auto-scaled display; mismatches (`8MB + 2s`) fail silently
- **Currencies**: markers as typed (`$`, `€`, `£`, `¥`, `kr`, any ISO code) with strict matching — no exchange-rate assumptions; `$ = USD`-style alias lines assign notation per note
- **Conversions**: `8MB in KiB`, `90min in h`, `255 in hex`, `Price in NOK/GB`
- Functions (trig in degrees), constants `pi`/`e` (shadowable), `#` comments
- Anything that doesn't fully parse **and** evaluate is left completely untouched — `Deadline = Friday` stays prose; headings, code fences, checkboxes, and tags are unaffected

## Implementation

- `src/utils/mathEvaluator.ts` — pure tokenizer → recursive-descent parser → dimensional evaluator → formatter; zero deps, no CM imports
- `src/components/editor/mathPlugin.ts` — CM6 ViewPlugin following the tagPlugin/imagePlugin patterns: `Decoration.mark` for variable names/refs, `Decoration.widget` (`side: 1`, `ignoreEvent`) for ghost results; derive-and-render only, no dispatches, undo history untouched; works on desktop and iOS
- Adds **vitest** as the repo's first test setup (`npm test`): 74 unit tests over the evaluator grammar and edge cases

## Verification

- 74 unit tests green; `tsc -b` and lint clean (the SettingsModal lint error pre-exists on main)
- Rendered-surface checks in headless Chromium: mounted the real plugin in a real EditorView, verified decorations, live recompute on edit, doc-integrity (decorations never mutate text), and coexistence with tag highlighting
- Also adds `.claude/skills/verify/SKILL.md` documenting the headless verification recipe

### CODE REVIEW

File: src/utils/mathEvaluator.ts [hunk 1]
```diff
@@ -0,0 +1,1036 @@
...
+
+  let inFence = false;
+  for (let i = 0; i < lines.length; i++) {
+    if (i <= skipThrough) continue;
+    const line = lines[i];
+    if (line.startsWith('```')) {
+      inFence = !inFence;
+      continue;
+    }
+    if (inFence || line.includes('`')) continue;
```
[the remark this answers is not in the document]
Comment (i-doll) [reply src/utils/mathEvaluator.ts]: Fixed in be09a8b — both ``` and ~~~ now open a fence, and only the matching delimiter closes it (so a ~~~ line inside a backtick fence stays content). Regression tests added for both cases.

File: src/utils/mathEvaluator.ts [hunk 2]
```diff
@@ -0,0 +1,1036 @@
...
+      }
+    }
+
+    // Trailing '=' means "show me the result".
+    let resultOffset: number | undefined;
+    const last = tokens[tokens.length - 1];
+    if (last.type === 'op' && last.op === '=') {
+      tokens.pop();
+      resultOffset = last.to;
+    }
```
[the remark this answers is not in the document]
Comment (i-doll) [reply src/utils/mathEvaluator.ts:962]: Leaving as-is: `tokenize()` creates a fresh array per line that is never shared or cached, and the in-place pop keeps the per-keystroke path allocation-free. Worth revisiting only if token arrays ever get cached.

File: src/utils/mathEvaluator.ts [hunk 3]
```diff
@@ -0,0 +1,1036 @@
...
+  for (const [name, value] of env) quantities.set(name, { value, dim: DIMLESS, code: null });
+  try {
+    const { node } = new Parser(tokens).parseTop();
+    const result = evalNode(node, quantities);
+    return Number.isFinite(result.value) ? result.value : null;
+  } catch (error) {
+    if (error instanceof MathError) return null;
+    throw error;
+  }
+}
```
[the remark this answers is not in the document]
Comment (i-doll) [reply src/utils/mathEvaluator.ts]: Removed in be09a8b — it was leftover public API from an earlier test plan; the tests exercise `analyzeDocument` directly.

File: src/components/editor/mathPlugin.ts [hunk 4]
```diff
@@ -0,0 +1,128 @@
...
+// above the viewport must stay in scope — same full-scan trade-off the tag
+// and link plugins already make. Notes are small; if a pathological note ever
+// matters, cache parses per line text.
+function mathDecorations(view: EditorView): DecorationSet {
+  const doc = view.state.doc;
+  const lines: string[] = [];
+  for (let i = 1; i <= doc.lines; i++) {
+    lines.push(doc.line(i).text);
+  }
+
```
[the remark this answers is not in the document]
Comment (i-doll) [reply src/components/editor/mathPlugin.ts:48]: Agreed — it's a documented trade-off (matches the existing tagPlugin/linkPlugin full-scan pattern, and whole-document scope is required for definitions above the viewport). A per-line parse cache keyed by line text is the noted escape hatch if large notes ever hurt.
